lib/dnsimple/client.rb

```diff
@@ -70,53 +70,57 @@ def base_url
     # Make a HTTP GET request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def get(path, options = {})
-      execute :get, path, options
+      execute :get, path, {}, options
     end
 
     # Make a HTTP POST request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def post(path, options = {})
-      execute :post, path, options
+    def post(path, data, options= {})
+      execute :post, path, data, options
     end
 
     # Make a HTTP PUT request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def put(path, options = {})
-      execute :put, path, options
+    def put(path, data, options = {})
+      execute :put, path, data, options
     end
 
     # Make a HTTP PATCH request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def patch(path, options = {})
-      execute :patch, path, options
+    def patch(path, data, options = {})
+      execute :patch, path, data, options
     end
 
     # Make a HTTP DELETE request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def delete(path, options = {})
-      execute :delete, path, options
+      execute :delete, path, {}, options
```

Comment:
Since we are not accepting attributes in the delete methods we would have to change all delete calls from something like:

```
response = client.delete(Client.versioned("/%s/contacts/%s" % [account_id, contact_id]), options)
```

to (note the extra `{}` in place of attributes):

```
response = client.delete(Client.versioned("/%s/contacts/%s" % [account_id, contact_id]), {}, options)
```

Either that, or add the attributes to each delete endpoint switching:

```
def delete_contact(account_id, contact_id, options = {})
```
to

```
def delete_contact(account_id, contact_id, attributes = {}, options = {})
```